tsl/test/sql/hypertable_generalization.sql

```diff
@@ -0,0 +1,241 @@
+SELECT create_hypertable('jobs_serial', by_range('job_id', partition_interval => 30));
+
+-- Insert data
+INSERT INTO jobs_serial (device_id, start_time, end_time)
+SELECT abs(timestamp_hash(t::timestamp)) % 10, t, t + INTERVAL '1 day'
+FROM generate_series('2018-03-02 1:00'::TIMESTAMPTZ, '2018-03-08 1:00':: TIMESTAMPTZ,'1 hour')t;
+
+-- Verify chunk pruning
+EXPLAIN VERBOSE SELECT * FROM jobs_serial WHERE job_id < 30;
+EXPLAIN VERBOSE SELECT * FROM jobs_serial WHERE job_id >= 30 AND job_id < 90;
+EXPLAIN VERBOSE SELECT * FROM jobs_serial WHERE job_id > 90;
+
+-- Update rows
+UPDATE jobs_serial SET end_time = end_time + INTERVAL '1 hour' where job_id = 1;
+UPDATE jobs_serial SET end_time = end_time + INTERVAL '1 hour' where job_id = 30;
+UPDATE jobs_serial SET end_time = end_time + INTERVAL '1 hour' where job_id = 90;
+
+SELECT start_time, end_time FROM jobs_serial WHERE job_id = 1;
+SELECT start_time, end_time FROM jobs_serial WHERE job_id = 30;
+SELECT start_time, end_time FROM jobs_serial WHERE job_id = 90;
+
+-- Delete rows
+DELETE FROM jobs_serial WHERE job_id < 10;
+
+SELECT count(*) FROM jobs_serial WHERE job_id < 30;
+
+DROP TABLE jobs_serial;
+
+-- Create and validate hypertable with BIGSERIAL column
+CREATE TABLE jobs_big_serial (job_id BIGSERIAL, device_id INTEGER, start_time TIMESTAMPTZ, end_time TIMESTAMPTZ, PRIMARY KEY (job_id));
+SELECT create_hypertable('jobs_big_serial', by_range('job_id', 100));
+
+-- Insert data
+INSERT INTO jobs_big_serial (device_id, start_time, end_time)
+SELECT abs(timestamp_hash(t::timestamp)) % 10, t, t + INTERVAL '1 day'
+FROM generate_series('2018-03-02 1:00'::TIMESTAMPTZ, '2018-03-08 1:00'::TIMESTAMPTZ,'30 mins')t;
+
+-- Verify #chunks
+SELECT count(*) FROM timescaledb_information.chunks;
+
+-- Get current sequence and verify updating sequence
+SELECT currval(pg_get_serial_sequence('jobs_big_serial', 'job_id'));
+
+-- Update sequence value to 500
+SELECT setval(pg_get_serial_sequence('jobs_big_serial', 'job_id'), 500, false);
+
+-- Insert few rows and verify that the next sequence starts from 500
+INSERT INTO jobs_big_serial (device_id, start_time, end_time)
+SELECT abs(timestamp_hash(t::timestamp)) % 10, t, t + INTERVAL '1 day'
+FROM generate_series('2018-03-09 1:00'::TIMESTAMPTZ, '2018-03-10 1:00'::TIMESTAMPTZ,'30 mins')t;
+
+-- No data should exist for job_id >= 290 to job_id < 500
+SELECT count(*) FROM jobs_big_serial WHERE job_id >= 290 AND job_id < 500;
+
+-- The new rows should be added with job_id > 500
+SELECT count(*) from jobs_big_serial WHERE job_id > 500;
+
+-- Verify show_chunks API
+SELECT show_chunks('jobs_big_serial', older_than => 100);
+SELECT show_chunks('jobs_big_serial', newer_than => 200, older_than => 300);
+SELECT show_chunks('jobs_big_serial', newer_than => 500);
+
+-- Verify drop_chunks API
+SELECT count(*) FROM timescaledb_information.chunks WHERE hypertable_name =
+'jobs_big_serial';
```

Comment:
Nit: Is this line break intentional? We have much longer lines without line breaks in this file. Also the following tests do not include the line break.